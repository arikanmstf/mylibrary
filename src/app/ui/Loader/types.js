/**
 * Component Types By => create-module script
 * @version 1.0.0
 *
 */

// @flow
export type LoaderProps = {
  isVisible: boolean,
};

export type LoaderContainerProps = {};
