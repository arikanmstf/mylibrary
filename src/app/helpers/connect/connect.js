// @flow
import React, { ComponentType } from 'react';

const connect = (mapUiToProps: Function) => (
  (WrappedComponent: ComponentType<*>) => {
    const uiComponents = mapUiToProps();
    return ({
      location, navigation, match, ...other
    }: Object) => (
      <WrappedComponent
        {...uiComponents}
        {...other}
        navigation={navigation}
        location={location || { pathname: navigation ? navigation.state.routeName : null }}
        match={match || { params: { id: navigation ? navigation.getParam('id') : undefined } }}
      />
    );
  }
);

export default connect;
