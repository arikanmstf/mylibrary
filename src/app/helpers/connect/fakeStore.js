class Store {
  constructor(props) {
    this.props = props;
  }

  toJS() {
    return this.props;
  }
}

class FakeStore {
  constructor(props) {
    this.props = props;
  }

  getState() {
    return new Store(this.props);
  }

  subscribe() {}
}

export const createFakeStore = (data) => {
  return new FakeStore(data);
};

export const dispatch = (action) => (action && typeof action === 'function' ? action() : undefined);
