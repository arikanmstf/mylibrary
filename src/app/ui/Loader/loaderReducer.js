/**
 * Reducers Template By => create-module script
 * @version 1.0.0
 *
 */

import { handleActions } from 'redux-actions';
import { showLoader, hideLoader } from './actions';

const initialState = {
  isVisible: false,
};

const reducer = new Map([
  [
    showLoader,
    (state) => ({ ...state, isVisible: true }),
  ],
  [
    hideLoader,
    (state) => ({ ...state, isVisible: false }),
  ],
]);

export default handleActions(
  reducer,
  initialState
);
