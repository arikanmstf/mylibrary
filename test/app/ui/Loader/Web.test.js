/**
 * Web Component Test Template By => create-module script
 * @version 1.0.0
 *
 */

import React from 'react';
import { shallow } from 'enzyme';
import Loader from 'ui/ModalLoader/Web';
import { createFakeStore } from 'helpers/connect/fakeStore';

describe('test/app/ui/Loader/Web.test.js', () => {
  it('Render', () => {
    const props = {};
    const store = createFakeStore({ loader: { isVisible: false } });
    const wrapper = shallow(<Loader {...props} store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
