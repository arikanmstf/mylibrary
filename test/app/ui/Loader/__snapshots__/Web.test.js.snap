// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/app/ui/Loader/Web.test.js Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Modal)
    store={
      FakeStore {
        "props": Object {
          "loader": Object {
            "isVisible": false,
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "dispatch": undefined,
      "isVisible": false,
      "store": FakeStore {
        "props": Object {
          "loader": Object {
            "isVisible": false,
          },
        },
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": FakeStore {
          "props": Object {
            "loader": Object {
              "isVisible": false,
            },
          },
        },
        "unsubscribe": undefined,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "dispatch": undefined,
        "isVisible": false,
        "store": FakeStore {
          "props": Object {
            "loader": Object {
              "isVisible": false,
            },
          },
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": FakeStore {
            "props": Object {
              "loader": Object {
                "isVisible": false,
              },
            },
          },
          "unsubscribe": undefined,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
